torch/export/export_schema.py

```diff
@@ -0,0 +1,326 @@
+from dataclasses import dataclass
+from enum import auto, Enum
+from typing import List, Union
+
+################################################################################
+# Following section is the defining a few utility data structures
+
+# As many of the serialization library doesn't have the concept of map
+# We use a list of sting-string pair to represent a map<string, string>
+@dataclass
+class StringStringPair:
+    key: str
+    value: str
+
+################################################################################
+# Following section is the defining the permissible argument types for operators
+
+# Copied from torchgen/model.py
+# !!! Consider if we want to add aliased dtypes, e.g float == float32. (Sherlock: No!)
+class ScalarType(Enum):
+    Byte = auto()
+    Char = auto()
+    Short = auto()
+    Int = auto()
+    Long = auto()
+    Half = auto()
+    Float = auto()
+    Double = auto()
+    ComplexHalf = auto()
+    ComplexFloat = auto()
+    ComplexDouble = auto()
+    Bool = auto()
+    BFloat16 = auto()
+
+# Copied from torch/_C/__init__.pyi.in
+# !!! Consider if we want to keep all of them, as we figures out what to do with SparseTensor
+class Layout(Enum):
+    # Defined in torch/csrc/utils/tensor_layouts.cpp
+    strided = auto()
+    sparse_coo = auto()
+    sparse_csr = auto()
+    sparse_csc = auto()
+    sparse_bsr = auto()
+    sparse_bsc = auto()
+    _mkldnn = auto()
+
+
+# Copied from torch/_C/__init__.pyi.in
+class MemoryFormat(Enum):
+    # Defined in torch/csrc/utils/tensor_memoryformats.cpp
+    contiguous_format = auto()
```

Comment:
Aten dialect demands all tensor be contiguous, therefore this field is redundant

Reply:
MemoryFormat is an permissible ArguementType, so it's still needed. 

E.g. aten.clone(t, memory_format = torch.contiguous_format)

Reply:
@qihqi that's only true if you convert all views to copies right? If functionalization preserves views (which it is allowed to do) you will have non-contiguous tensors

Reply:
Could you elaborate on "preserve views"? Is it about when input tensors are "views" themselves? Or are there cases where functionalization doesn't convert a view op into its copy counterpart?

@bdhirsh

Reply:
We ( @suo @ezyang ) discussed this today. We said currently we're doing views to copies, and thus we assumed all to be contiguous, but that we're not committed to not changing this so I think it's fine to reflect with what for now seems redundant field.

Reply:
Does it means that here shouldn't be padding for say 7x7 byte tensor?